build_tools/bump_submodules.py

```diff
@@ -52,7 +52,74 @@ def pin_ck():
     )
 
 
-def run(args):
+def parse_components(components: list[str]) -> list[list]:
+    arguments = []
+    system_projects = []
+
+    # If `default` is passed, use the defaults set in `fetch_sources.py` by not passing additonal arguments
+    if "default" in components:
+        return [], []
+
+    if any(comp in components for comp in ["base", "comm-libs", "core", "profiler"]):
+        arguments.append("--include-system-projects")
+    else:
+        arguments.append("--no-include-system-projects")
+
+    if "base" in components:
+        system_projects += [
+            "half",
+            "rocm-cmake",
+            "rocm-core",
+            "rocm_smi_lib",
+            "rocprofiler-register",
+        ]
+
+    if "comm-libs" in components:
+        system_projects += [
+            "rccl",
+            "rccl-tests",
+        ]
+
+    if "core" in components:
+        # TODO: Add "amdgpu-windows-interop"
+        system_projects += [
+            "HIP",
+            "ROCR-Runtime",
+            "clr",
+            "rocminfo",
+        ]
```

Comment:
Any particular reason to not include amdgpu-windows-interop now? The repository only changes when we push to it right now, so it shouldn't add much cost here I think.

Reply:
It's explicitly excluded in `fetch_sources`
https://github.com/ROCm/TheRock/blob/7b4a3fdcc7fd32a46f4a3bf1891d4a91b1b88a8e/build_tools/fetch_sources.py#L334-L340
if not on Windows, thus I wasn't sure we want to add it here. But we probably can just add it here even though it would normally not be fetched on Linux.

Reply:
Ah, right. I'd either put the same condition in this file or change from "TODO: add it" to "amdgpu-windows-interop is Windows only and is updated manually"